rpcserver.go

```diff
@@ -6207,15 +6260,51 @@ func (r *rpcServer) FundingStateStep(ctx context.Context,
 			return nil, err
 		}
 
+	// There is no need to register a PSBT shim before opening the channel,
+	// even though our RPC message structure allows for it. Inform the user
+	// by returning a proper error instead of just doing nothing.
+	case in.GetShimRegister() != nil &&
+		in.GetShimRegister().GetPsbtShim() != nil:
```

Comment:
No, I don't see a use case where this would be needed. This is added to catch incorrect API usage and inform the user they're doing something wrong instead of just returning an empty message.